naptime/src/main/scala/org/coursera/naptime/Types.scala

```diff
@@ -58,6 +64,19 @@ object Types extends StrictLogging {
       case unknown: DataSchema =>
         throw new RuntimeException(s"Cannot compute asymmetric type for key type: $unknown")
     }
+    for ((name, relation) <- fields.relations) {
+      Option(mergedSchema.getField(name)) match {
+        case None =>
+          logger.warn(s"Fields for resource $typeName mentions field name '$name' but this field " +
+            "is not found in the merged type schema.")
+        case Some(field) =>
+          val properties = field.getProperties.asScala
+          val relatedMap = Map[String, AnyRef](
+            Relations.PROPERTY_NAME -> relation.identifier)
```

Comment:
do we validate these identifiers anywhere (ensuring they're at least a valid resource name)?